third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -3683,6 +3683,10 @@ impl ExpTranslator<'_, '_, '_> {
         }
 
         if let Some(entry) = self.parent.parent.fun_table.get(&global_var_sym) {
+            if entry.kind == FunctionKind::Inline {
```

Comment:
Could we please add the following tests (separating out the positive and negative tests in different files):

```
    public inline fun foo(x: u64): u64 {
        x + 1
    }

    public fun negative(): u64 {
        let f = foo;
        f(42)        
    }

    public fun positive_1(): u64 {
        let f = || foo(42);
        f()
    }

    public fun positive_2(): u64 {
        let f = |x| foo(x);
        f(42)
    }
```

Reply:
added the tests.